internal/files/files.go

```diff
@@ -71,3 +72,28 @@ func FileType(fs vfs.FileSystem, path string) (string, error) {
 	// content-type by returning "application/octet-stream" if no others seemed to match.
 	return http.DetectContentType(buf), nil
 }
+
+// FindDirectoryContaining returns the path to the directory containing the targetFolderName
+// and a bool indicating if the directory was found or not.
+// If the directory was not found, the error will be os.ErrNotExist.
+// The search starts at the path and continues to the root of the filesystem.
+// The targetFolderName can be a relative path.
+// Example:
+//
+//	FindDirectoryContaining("` + path + `", "` + targetFolderName + `")
+func FindDirectoryContaining(path, targetFolderName string) (string, bool, error) {
+	if path == string(filepath.Separator) {
+		return "", false, os.ErrNotExist
+	}
+
+	targetPath := filepath.Join(path, targetFolderName)
+	_, err := os.Stat(targetPath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return FindDirectoryContaining(filepath.Dir(path), targetFolderName)
```

Comment:
Why do we need recursive lookup here? The required path flag of the command already points to either an absolute or relative path or?